python/tank/platform/environment.py

```diff
@@ -346,19 +346,28 @@ def get_app_descriptor_dict(self, engine_name, app_name):
                             "key for app %s.%s" % (self._env_path, engine_name, app_name))
         return descriptor_dict
 
-    def find_location_for_engine(self, engine_name):
+    def find_location_for_engine(self, engine_name, absolute_location=False):
```

Comment:
could we contain the change so that it is internal within `environment.py`? Do we have to expose the `absolute_location` flag to the rest of the world? Seems internal to me.

Reply:
Note: The motivation for this is that we want to be able to put the environment data intoa database or some other file structure - the interface around the environment access should not be coupled with how it is being stored or now things are interconnected via `@includes` in those files.

Reply:
As discussed, I totally agree. :)